File: 201-300/201-210/205-isomorphicStrings/isomorphicStrings.py
class Solution(object):
    def isIsomorphic(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """

        dictionary, dictionary2 = {}, {}
        length = len(s)

        if length != len(t):
            return False

        for i in range(length):
            lchar, rchar = s[i], t[i]
            dictionary.setdefault(lchar, rchar)
            dictionary2.setdefault(rchar, lchar)

            if dictionary[lchar] != rchar or dictionary2[rchar] != lchar:
                return False

        return True

File: 201-300/201-210/205-isomorphicStrings/isomorphicStrings.cpp
#include <unordered_map>

using namespace std;

class Solution {
public:
    bool isIsomorphic(string s, string t) {
        unordered_map<char, char> stot;
        unordered_map<char, char> ttos;

        for (int i = 0; i < s.size(); ++i) {
            stot.emplace(s[i], t[i]);
            ttos.emplace(t[i], s[i]);

            if (stot[s[i]] != t[i] || ttos[t[i]] != s[i])
                return false;
        }

        return true;
    }
};